internal/api/token_refresh.go

```diff
@@ -36,7 +36,7 @@ func (a *API) RefreshTokenGrant(ctx context.Context, w http.ResponseWriter, r *h
 		return oauthError("invalid_request", "refresh_token required")
 	}
 
-	user, token, session, err := models.FindUserWithRefreshToken(db, params.RefreshToken)
+	user, _, session, err := models.FindUserWithRefreshToken(db, params.RefreshToken, false)
```

Comment:
can we just push this and the check for whether a user is banned down to the transaction since we're also querying for the user there?

Reply:
It's intentional. We don't want to lock on the session if it is already expired or the token is missing. We only want to lock after the basic checks have passed, which is why I added the comment.